.github/workflows/codeql.yml

```diff
@@ -0,0 +1,25 @@
+name: CodeQL Analysis
+
+on:
+  workflow_call:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    timeout-minutes: 120
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          show-progress: false
```

Comment:
zizmor will probably complain about not having persist-credentials: false here, which stops later steps having access to the checkout's credentials.